hs_access_control/migrations/0015_manual_remove_redundant_privileges.py

```diff
@@ -0,0 +1,156 @@
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals
+from django.db import migrations, models
+
+
+def remove_extra_privileges(apps, schema_editor):
+    """ 
+    Remove extra records from privileges tables
+        
+    This change removes extra records that were generated by early versions of the user interface. 
+    These records are redundant, in the sense that the highest privilege always wins. 
+    The method for eliminating them is to retain the highest privilege that was instituted latest. 
+    This is consistent with what would have resulted if the access control system were running 
+    under the new semantics. 
+
+    For example, if user 'foo' had the following privilege records over resource 'bar': 
+    
+    
+        VIEW   1/1/2016  michael 
+        CHANGE 1/2/2016  robert
+        VIEW   2/1/2016  alva
+
+    Then 
+    
+    * The new effective privilege record will be: 
+        
+        CHANGE 1/2/2016  robert
+
+    * The first record will be considered to be overridden by the second. 
+    * The third record cannot be honored because it would result in a perceivable downgrade of 
```

Comment:
I assume in the current system, we allow downgrading of privilege. Then if 'alva' downgraded 'foo's' privilege from CHANGE to VIEW why that should not be honored during migration? 

I would think the deletion of privilege records be based on the logic that we just keep the privilege that was granted last.